src/main/scala/com/databricks/spark/csv/util/InferSchema.scala

```diff
@@ -53,14 +55,20 @@ private[csv] object InferSchema {
     StructType(structFields)
   }
 
-  private def inferRowType(nullValue: String, dateFormatter: SimpleDateFormat)
+  private def inferRowType(nullValue: String,
```

Comment:
How does this function work? As far as I can see it is passed the flag for `dropMalformed` which is constant for all the rows (either `true` or `false`). Shouldn't the flag be set based on if the row is malformed or not?